modules/apps/frontend-taglib/frontend-taglib-clay/src/main/resources/META-INF/resources/data_set_display/DataSetDisplay.js

```diff
@@ -257,39 +257,36 @@ function DataSetDisplay({
 		}
 	}, [wrapperRef]);
 
-	const refreshData = useCallback(
-		(successNotification) => {
-			setLoading(true);
-
-			return requestData()
-				.then((data) => {
-					const {
-						message,
-						showSuccessNotification,
-					} = successNotification;
-
-					if (showSuccessNotification) {
-						openToast({
-							message:
-								message ||
-								Liferay.Language.get('table-data-updated'),
-							type: 'success',
-						});
-					}
+	function refreshData(successNotification) {
+		setLoading(true);
+
+		return requestData()
+			.then((data) => {
+				if (successNotification?.showSuccessNotification) {
+					openToast({
+						message:
+							successNotification.message ||
+							Liferay.Language.get('table-data-updated'),
+						type: 'success',
+					});
+				}
 
-					if (isMounted()) {
-						setLoading(false);
-						updateDataSetItems(data);
+				if (isMounted()) {
+					setLoading(false);
+					updateDataSetItems(data);
 
-						Liferay.fire(DATASET_DISPLAY_UPDATED, {id});
-					}
+					Liferay.fire(DATASET_DISPLAY_UPDATED, {id});
+				}
 
-					return data;
-				})
-				.catch(() => setLoading(false));
-		},
-		[id, isMounted, requestData]
-	);
+				return data;
+			})
+			.catch((error) => {
+				logError(error);
+				setLoading(false);
+
+				throw error;
```

Comment:
Hey @FabioDiegoMastrorilli, I think `logError` should be enough. But we can remove this later.